src/main/java/by/onliner/utils/Recorder.java

```diff
@@ -0,0 +1,122 @@
+     * @param audioFormat audio format
+     * @param movieFolder movie file
+     * @param name method name
+     */
+    public Recorder(GraphicsConfiguration cfg, Rectangle captureArea, Format fileFormat,
+                    Format screenFormat, Format mouseFormat, Format audioFormat, File movieFolder, String name)
+            throws IOException, AWTException {
+        super(cfg, captureArea, fileFormat, screenFormat, mouseFormat, audioFormat, movieFolder);
+        SimpleDateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd HH.mm.ss");
+        Recorder.name = name + "-" + dateFormat.format(new Date()) + "." + Registry.getInstance().getExtension(fileFormat);
+    }
+
+    /***
+     * Creates a file for recording the movie.
+     * @param fileFormat format of the file
+     * @return created file
+     */
+    @Override
+    protected File createMovieFile(Format fileFormat) throws IOException {
+        if (!movieFolder.exists()) {
+            movieFolder.mkdirs();
+        } else if (!movieFolder.isDirectory()) {
+            throw new IOException(String.format("{%s} is not a directory.", movieFolder));
+        }
+        return new File(movieFolder, Recorder.name);
+    }
+
+    /***
+     * Initialise recorder and start recording test
+     * @param name method name
+     */
+    public static void startRecording(String name) throws IOException, AWTException {
+        File file = new File(SAVE_VIDEO_PATH);
+
+        Dimension screenSize = Toolkit.getDefaultToolkit().getScreenSize();
+        int width = screenSize.width;
+        int height = screenSize.height;
+
+        Rectangle captureSize = new Rectangle(0, 0, width, height);
+
+        GraphicsConfiguration gc = GraphicsEnvironment.getLocalGraphicsEnvironment().
+                getDefaultScreenDevice()
+                .getDefaultConfiguration();
+
+        screenRecorder = new Recorder(gc, captureSize,
+                new Format(MediaTypeKey, MediaType.FILE, MimeTypeKey, MIME_AVI),
+                new Format(MediaTypeKey, MediaType.VIDEO, EncodingKey, ENCODING_AVI_TECHSMITH_SCREEN_CAPTURE,
+                        CompressorNameKey, ENCODING_AVI_TECHSMITH_SCREEN_CAPTURE, DepthKey, 24, FrameRateKey,
+                        Rational.valueOf(15), QualityKey, 1.0f, KeyFrameIntervalKey, 15 * 60),
+                new Format(MediaTypeKey, MediaType.VIDEO, EncodingKey, "black", FrameRateKey, Rational.valueOf(30)),
+                null, file, name);
+        // This magical numbers from indian guy, don't touch!
+
+        screenRecorder.start();
+    }
+
+    /***
+     * Stoped recording test
+     * @param name method name
+     */
+    public static void stopRecording(String name) throws IOException {
+        screenRecorder.stop();
+        aviToMp4(name);
+    }
+
+    /***
+     * Converts avi to mp4 video format
+     * @param name file name
+     */
+    private static void aviToMp4(String name) {
+        File source = new File(SAVE_VIDEO_PATH + Recorder.name);
```

Comment:
What OOP principe is broken by accessing Recorder.name ?

Reply:
Fixed